queries.md

```diff
@@ -34,16 +34,16 @@
 
 ## list orders grouped by customer showing the number of orders per customer. _Rattlesnake Canyon Grocery_ should have 7 orders.
 
-- 
+- SELECT Customers.CustomerName, Count(*) FROM [Orders] JOIN Customers ON Orders.CustomerID=Customers.CustomerID GROUP BY CustomerName;
 
 ## list customers names and the number of orders per customer. Sort the list by number of orders in descending order. _Ernst Handel_ should be at the top with 10 orders followed by _QUICK-Stop_, _Rattlesnake Canyon Grocery_ and _Wartian Herkku_ with 7 orders each.
 
-- 
+- SELECT Customers.CustomerName, Count(*) FROM [Orders] JOIN Customers ON Orders.CustomerID=Customers.CustomerID GROUP BY CustomerName ORDER BY Count(*) desc;
 
 ## list orders grouped by customer's city showing number of orders per city. Returns 58 Records with _Aachen_ showing 2 orders and _Albuquerque_ showing 7 orders.
 
-- 
+- SELECT Customers.City, COUNT(*) FROM [Orders] JOIN [Customers] WHERE Customers.CustomerId = Orders.CustomerId GROUP BY City;
 
 ## delete all users that have no orders. Should delete 17 records.
 
-- 
\ No newline at end of file
+- DELETE FROM [Customers] WHERE CustomerID NOT IN (SELECT Customers.CustomerID FROM [Orders] JOIN Customers ON Orders.CustomerID=Customers.CustomerID GROUP BY CustomerName);
```

Comment:
With big lines like these, we usually split them up by keyword:
```sql
DELETE 
  FROM customers
  WHERE customerId
  NOT IN (
    SELECT c.customerId
      FROM orders as o
      JOIN customers as c
      ON o.customerId = c.customerId
      GROUP BY customerName
  )
```

Reply:
Note the use of aliases here; they're quite useful when it comes to joining two tables with fields that have the same name.